vault-util/cmd/fetch-secret-env.go

```diff
@@ -0,0 +1,97 @@
+package cmd
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"log"
+	"os"
+
+	"github.com/spf13/cobra"
+
+	"github.com/chef/ci-studio-common/lib"
+	"github.com/chef/ci-studio-common/vault-util/internal/vault"
+)
+
+var (
+	fetchSecretsCmd = &cobra.Command{
+		Use:   "fetch-secret-env",
+		Short: "Fetch the secrets specified in the VAULT_UTIL_SECRETS environment variable from Vault.",
+		Run:   printSecrets,
+	}
+
+	fetchSecretsOpts = struct {
+		format string
+	}{}
+
+	vaultClient *vault.Client
+	secretsJSON map[string]interface{}
+)
+
+func init() {
+	var err error
+
+	vaultClient, err = vault.NewClient()
+	lib.Check(err)
+
+	rootCmd.AddCommand(fetchSecretsCmd)
+
+	fetchSecretsCmd.Flags().StringVar(&fetchSecretsOpts.format, "format", "sh", "Which format to use when exporting environment variables.")
+}
+
+func printSecrets(cmd *cobra.Command, args []string) {
+	secretsEnv := os.Getenv("VAULT_UTIL_SECRETS")
+
+	if len(secretsEnv) == 0 {
+		return
+	}
+
+	err := json.Unmarshal([]byte(secretsEnv), &secretsJSON)
+	lib.Check(err)
+
+	for k, v := range secretsJSON {
+		value := fetchSecret(v.(map[string]interface{}))
+
+		switch fetchSecretsOpts.format {
+		case "sh":
+			fmt.Printf("export %s=%q\n", k, value)
+		case "ps1":
+			fmt.Printf("$env:%s=%q\n", k, value)
+		case "batch":
+			fmt.Printf("\"%s=%s\"\n", k, value)
+		default:
+			log.Fatalf("'%s' is not a supported format", fetchSecretsOpts.format)
+		}
+	}
+}
+
+func fetchSecret(settings map[string]interface{}) string {
+	if value, exists := settings["value"]; exists {
```

Comment:
We might not know the top-level keys, but we know the leaves structures; so, to avoid failing type assertions, we could define
```go
type secret struct {
  Field string `json:"field"`
  Path string `json:"path"`
  Value string `json:"value"`
  Account string `json:"account"
}
```
and 
```go
secretsJSON map[string]secret
```
so we could write this as
```go
func fetchSecret(s secret) string {
	if s.value != "" {
		return s.value
	}
	if s.field == "" {
		log.Fatal("missing required 'field' setting") // note: don't think we need the errors.New here
	}

	// etc
}
```
It might not be a big win in terms of error checking, but anytime we write `x.(string)`, there's a panic lurking behind it 😉